### PR TITLE
fix: mathjax bug add client provider

### PR DESCRIPTION
**Pull request by JBR90**
## Description

- Moves mathjax context to provider, explicitly states client function in next
- Moves provider to wrap just quizzes rather than page

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2043--oak-web-application.netlify.thenational.academy/teachers/programmes/maths-secondary-ks4-higher-l/units/adding-surds-a57d/lessons/subtract-two-surds-6njkac#starter-quiz
2. Click on the surds lessons - mathjax should still render with no errors

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

**Comment by njblore** [comment]
Oh! I got the error on the preview! but the maths loaded fine so ... bad news is this doesn't seem to fix it? good news is there was no impact on the user 🤷‍♀️ so I'm not sure what to do with this 
  
Screenshot 2023-10-24 at 11 41 00

**Review by rightsaidjames — APPROVED** [review]
Functionally this seems fine, and I'm not seeing the error that @njblore highlighted either. Checked some 2023 content as well, plus some non-Maths content - all fine.

File: src/components/QuizContainerNew/QuestionsListNew/QuestionListItemNew/QuestionListItem.tsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ const QuestionListItem: FC<QuestionListItemProps> = (props) => {
   const { questionStem, answers } = question;
 
   return (
-    <MathJax hideUntilTypeset="every" dynamic>
+    <MathJax>
```
Comment (njblore) [inline src/components/QuizContainerNew/QuestionsListNew/QuestionListItemNew/QuestionListItem.tsx:23]: should this be using `MathJaxWrap` ?

Comment (JBR90) [reply src/components/QuizContainerNew/QuestionsListNew/QuestionListItemNew/QuestionListItem.tsx:23]: Yes we should!

**Review by njblore — APPROVED** [review]

**Pull request merged**